### ISSUE TITLE
Deadlock sometimes in "sync" function

### ISSUE THREAD

**Issue by huhn511** [iotaledger/wallet.rs#316]
## Bug description

We tried the nodejs binding  to write some examples. The program soemtimes not responding while using 

`    const synced = await account.sync()`

## Rust version

Which version of Rust are you running?

- Rust version: rustc 1.49.0 (e1884a8e3 2020-12-29
- nodejs version: v12.14.0

## Version

Which version of the library are you using?

Version
npm version 0.0.2 installed from npmjs.org

## Steps To reproduce the bug

Explain how the maintainer can reproduce the bug.
1. git clone https://github.com/huhn511/wallet.rs
2. git checkout feat(docs)/restore_example
3. cd bindings/nodejs/examples/
4. node 1-create-account.js
5.  node 2-generate-address.js

-> Run 5. multiple times

## Expected behaviour

Describe what you expect to happen.

Get the latest address

## Actual behaviour

Describe what actually happens.

Deadlock

## Errors

No error logs - just "not respoding"

**Comment by Thoralf-M** [comment]
I tried to get the balance in Rust and after calling `account.sync().await;` with an account that should have balance it still returns `AccountBalance { total: 0, available: 0, incoming: 0, outgoing: 0 }`
with `manager.sync_accounts().await?;` it works but returns, don't know if the panick is completely unrelated or not
```
Account balance: AccountBalance { total: 10000000, available: 10000000, incoming: 10000000, outgoing: 0 }
thread 'pool-thread-#3' panicked at 'called `Result::unwrap()` on an `Err` value: ReturnReadStore([119, 97, 108, 108, 101, 116, 45, 97, 99, 99, 111, 117, 110, 116, 58, 47, 47, 53, 101, 99, 102, 100, 48, 102, 55, 55, 56, 54, 52, 52, 57, 56, 101, 100, 52, 55, 56, 99, 57, 99, 54, 53, 98, 100, 101, 98, 48, 98, 57, 97, 54, 102, 99, 53, 51, 56, 52, 98, 52, 50, 54, 56, 101, 50, 56, 51, 54, 99, 51, 56, 48, 99, 102, 56, 49, 56, 51, 57, 100, 100, 49, 124], Ok(()))', .cargo\git\checkouts\stronghold.rs-fe28308a6634692d\87ec1eb\client\src\utils\ask.rs:```

**Comment by lucasfernog** [comment]
Fixed by https://github.com/iotaledger/iota.rs/pull/331

Let me know if you still face this issue :)
